compiler/sigmatch.nim

```diff
@@ -1425,6 +1425,10 @@ proc typeRelImpl(c: var TCandidate, f, aOrig: PType,
     considerPreviousT:
       if a.kind == tyGenericInst and a.sons[0] == f:
         bindingRet isGeneric
+      if a.kind == tyGenericBody:
```

Comment:
I think the previous version was better. Generic bodies act as nominal types and the direct pointer comparison is more appropriate. What Araq requested was just to change the return type to `isGeneric`.

Reply:
Araq told me `==` was to be avoided, that's why I added one more typerel pass to avoid it. But anyway if you think the plain old pointer comparison will work I'll revert the patch to its first version

Reply:
ok.

Reply:
I think it's ready now